modules/admin-ui/src/main/java/org/opencastproject/adminui/impl/MediaPackageLockService.java

```diff
@@ -0,0 +1,176 @@
+
+import static java.lang.Math.max;
+
+import org.opencastproject.index.service.impl.index.event.Event;
+import org.opencastproject.mediapackage.Publication;
+import org.opencastproject.mediapackage.Track;
+
+import org.json.simple.JSONObject;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Dictionary;
+import java.util.HashMap;
+
+/**
+ *
+ * @author Tobias M Schiebeck
+ */
+public class MediaPackageLockService implements ManagedService {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(MediaPackageLockService.class);
+
+  /**
+   * minimum duration the editor is locked to avoid synchronous editing
+   */
+  private static final long MIN_LOCK_DURATION = 1800000;
+
+  private static final HashMap<String, EditorLock> mpLock = new HashMap<String, EditorLock>();
+
+  /**
+   * The Json key for the MediaPackageId.
+   */
+  public static final String MEDIAPACKAGEID_KEY = "id";
+
+  /**
+   * The Json key for the SessionId.
+   */
+  public static final String SESSION_KEY = "session";
+
+  public MediaPackageLockService() {
+    super();
+    logger.info("starting MP Lock service");
+  }
+
+  public long getMediaPackageLock(Event event, String sessionId) {
+    EditorLock releaseLock = null;
+    EditorLock lock = new EditorLock(event.getIdentifier(), sessionId);
+    logger.debug("lock mp {} for session {}",event.getIdentifier(), sessionId);
+    synchronized (mpLock) {
+      releaseLock = mpLock.get(lock.getMediaPackageId());
+      if (releaseLock == null) {
+        Long lockDur = event.getDuration();
+        if (lockDur == null) {
+          lockDur = 0L;
+          for (Publication pub : event.getPublications()) {
+            for (Track track : pub.getTracks()) {
+              lockDur = max(lockDur, (long) track.getDuration());
+            }
+          }
+        }
+        lock.setDuration(lockDur);
+        mpLock.put(lock.getMediaPackageId(), lock);
+        return -1;
+      }
+      return releaseLock.checkLock(lock);
+    }
+  }
+
+  public void releaseMediaPackageLock(final Event event, String sessionId) {
+    logger.debug("unlock mp {}",event.getIdentifier());
+    synchronized (mpLock) {
+      EditorLock el = mpLock.get(event.getIdentifier());
+      if (el == null) {
+        for (String eventId : mpLock.keySet()) {
+          if (mpLock.get(eventId).removable(sessionId)) {
+            mpLock.remove(eventId);
+          }
+        }
+      } else {
+        if (el.removable(sessionId)) {
+          mpLock.remove(event.getIdentifier());
+        }
+      }
+    }
+  }
+
+  @Override
+  public void updated(Dictionary<String, ?> dctnr) throws ConfigurationException {
+    throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```

Comment:
Throwing an exception in a OSGI method should be avoid. If you are not using the `updated(Dictionary)` method, you should drop `ManagedService` implementation

Reply:
I need managed service otherwise I cannot register the service througg OSGI